File: Data Structures & Algorithms/max-water-container/submission-3.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        res , r = 0 , len(heights)-1
        l=0
        while l < r:
            h=min(heights[r],heights[l])
            res = max (h*(r-l),res)
            if heights[l] <= heights[r]:
                l += 1
            else:
                r -= 1
        return res

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-3.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int res=0;
        int mx=0,mn= prices[0];
        for(int price:prices){
            res=max(res, price-mn);
            mn=min(price,mn);
        }
        return res;
    }
};
